rice-middleware/kim/kim-api/src/main/java/org/kuali/rice/kim/api/group/GroupService.java

```diff
@@ -101,6 +101,21 @@
     List<Group> getGroupsByPrincipalIdAndNamespaceCode(@WebParam(name = "principalId") String principalId,
             @WebParam(name = "namespaceCode") String namespaceCode) throws RiceIllegalArgumentException;
 
+
+
+    /**
+     * Get a a flat list of groups and their parent groups based on group ids.
+     *
+     * <p>
+     * This will include all groups directly assigned as well as those inferred
+     * by the fact that they are members of higher level groups, and filtered by Group namespace.
+     * </p>
+     *
+     * @param groupIds List of group ids
+     * @return a list of Group objects
+     */
+    List<Group> getParentGroupsByGroupsIds(List<String> groupIds);
```

Comment:
We'll want to add the SOAP service annotations here so we can invoke this in a web service call.  Probably should also add `@since 2.4`